src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -40,6 +46,8 @@ const useCoreChatSystemHook = () => {
   const [migrationStatus] = useRecoilState(migrationDbStatusAtom);
   const initialStartup: any = useRecoilValue(InitialStartupAtom);
   const {params} = useRoute();
+  const [replyPreview, dispatch] = (React.useContext(Context) as unknown as any).chat;
+  const [processedMessageId, setProcessedMessageId] = React.useState<string | null>(null);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The context is being cast to `any` which can lead to potential type errors. It would be better to define a proper interface for the context and use that instead.
```diff
- const [replyPreview, dispatch] = (React.useContext(Context) as unknown as any).chat;
+ interface ChatContext {
+   chat: [ReplyPreviewType, Dispatch<ReplyPreviewActionType>];
+ }
+ 
+ const { chat: [replyPreview, dispatch] } = React.useContext(Context) as ChatContext;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->